### PR TITLE
risc-v/mpfs: serial: add fpga uarts

### PR DESCRIPTION
**Pull request by eenurkka**
Add FPGA serial support, including console capabilities.

## Summary

This adds a simple FPGA uart for testing

## Testing

Limited, > 1MB data from / to linux Ubuntu 22.04

### CODE REVIEW

File: arch/risc-v/src/mpfs/mpfs_serial.c [hunk 1]
```diff
@@ -330,6 +364,7 @@ static struct up_dev_s g_uart4priv =
   .parity    = CONFIG_UART4_PARITY,
   .bits      = CONFIG_UART4_BITS,
   .stopbits2 = CONFIG_UART4_2STOP,
+  .fpga      = false,
```
Comment (pussuw) [inline arch/risc-v/src/mpfs/mpfs_serial.c:367]: I think these can be fpga too. We won't use hw+fpga uarts at the same time. @jlaitine can confirm.

**Review by pussuw — APPROVED** [review]
LGTM

File: arch/risc-v/src/mpfs/mpfs_serial.c (hunk 1 above)
Comment (pussuw) [reply arch/risc-v/src/mpfs/mpfs_serial.c:367]: Maybe make this selectable via kconfig, so we can get the hw blocks working if needed.

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_serial.c:348]: Yes, I already started with some stuff, but decided it got too complicated. Single switch looks the best (until you find out you need one MSS UART :)  )

Comment (eenurkka) [reply arch/risc-v/src/mpfs/mpfs_serial.c:349]: Better now, even fpga uart0 works as a console. Waiting for the other uarts, offsets are just ok only for the FPGA UART0, others will be changed once we know what they are. Will add 2 more uarts (now only 6).

**Review by jlaitine — APPROVED** [review]
LGTM

**Pull request merged**